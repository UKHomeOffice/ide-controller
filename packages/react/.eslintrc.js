// @todo add vs code format on save wiki/ readme on github
module.exports = {
  env: {
    browser: true,
    es2020: true,
    node: true,
    jest: true,
  },
  extends: ['plugin:react/recommended', 'airbnb', 'prettier'],
  parserOptions: {
    ecmaFeatures: {
      jsx: true,
    },
    ecmaVersion: 11,
    sourceType: 'module',
  },
  plugins: ['react', 'prettier'],
  ignorePatterns: ['/src/opencv'],
  rules: {
    quotes: ['error', 'single'],
    'prettier/prettier': [
      'error',
      {
        trailingComma: 'es5',
        singleQuote: true,
        printWidth: 80,
      },
    ],
  },
};
