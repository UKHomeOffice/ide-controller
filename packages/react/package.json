{
  "name": "@ide-controller/react",
  "version": "1.6.0",
  "private": true,
  "homepage": "./",
  "author": "Viable Data",
  "scripts": {
    "dev": "BROWSER=none CI=true react-scripts start",
    "build": "react-scripts build",
    "test": "react-scripts test --watchAll=false",
    "test:watch": "react-scripts test",
    "eject": "react-scripts eject",
    "lint": "npx eslint --ext .js,.jsx .",
    "lint:fix": "eslint --ext .js,.jsx . --fix"
  },
  "browserslist": {
    "production": [
      ">0.2%",
      "not dead",
      "not op_mini all"
    ],
    "development": [
      "last 1 chrome version",
      "last 1 firefox version",
      "last 1 safari version"
    ]
  },
  "dependencies": {
    "@tensorflow-models/posenet": "^2.2.1",
    "@tensorflow/tfjs": "^2.7.0",
    "govuk-frontend": "^3.8.0",
    "prop-types": "^15.7.2",
    "react": "^16.13.1",
    "react-dom": "^16.13.1",
    "react-scripts": "^3.4.3"
  },
  "devDependencies": {
    "@testing-library/jest-dom": "^4.2.4",
    "@testing-library/react": "^9.3.2",
    "@testing-library/user-event": "^7.1.2",
    "eslint-config-airbnb": "^18.2.0",
    "eslint-config-prettier": "^6.11.0",
    "eslint-plugin-import": "^2.22.1",
    "eslint-plugin-prettier": "^3.1.4",
    "eslint-plugin-react": "^7.21.5",
    "jest-canvas-mock": "^2.3.0",
    "node-sass": "^4.14.1",
    "prettier": "^2.1.1"
  }
}
