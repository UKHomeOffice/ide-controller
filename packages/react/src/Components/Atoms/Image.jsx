// Global imports
import PropTypes from 'prop-types';
import React from 'react';

const Image = ({ image, imageAlt, className }) => {
  return (
    <div
      className={`photoContainer--photo ${className}`}
      alt={imageAlt}
      style={{ backgroundImage: `url(${image})` }}
      data-testid="atoms-image"
    />
  );
};

Image.propTypes = {
  image: PropTypes.string.isRequired,
  imageAlt: PropTypes.string,
  className: PropTypes.string,
};

Image.defaultProps = {
  imageAlt: '',
  className: '',
};

export default Image;
