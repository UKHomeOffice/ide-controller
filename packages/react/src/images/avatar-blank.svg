<?xml version="1.0" encoding="UTF-8"?>
<svg width="234px" height="271px" viewBox="0 0 234 271" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
    <title>avatar-blank</title>
    <g id="Page-1" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd" fill-opacity="0.1">
        <g id="avatar-blank" fill="#000000" fill-rule="nonzero">
            <circle id="Oval" cx="117" cy="70" r="70"></circle>
            <path d="M169,155 C204.898509,155 234,184.101491 234,220 L234,271 L0,271 L0,220 C0,184.101491 29.1014913,155 65,155 L169,155 Z" id="Combined-Shape"></path>
        </g>
    </g>
</svg>