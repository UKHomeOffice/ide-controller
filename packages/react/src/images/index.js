/* eslint import/prefer-default-export: "off" */
export { default as blankAvatar } from './avatar-blank.svg';
