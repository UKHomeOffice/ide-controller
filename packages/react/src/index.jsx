// Global imports
import React from 'react';
import ReactDOM from 'react-dom';
import { initAll } from 'govuk-frontend';
import 'govuk-frontend/govuk/all.scss';

// Local imports
import App from './App';
import './Components/Style/atoms.scss';
import './Components/Style/global.scss';
import './Components/Style/molecules.scss';
import './helpers/ipcMainEvents';

document.body.className = document.body.className
  ? `${document.body.className} js-enabled`
  : 'js-enabled';

initAll();

ReactDOM.render(
  <React.StrictMode>
    <App />
  </React.StrictMode>,
  document.getElementById('root')
);
