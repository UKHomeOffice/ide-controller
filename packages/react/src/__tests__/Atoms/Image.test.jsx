// Global imports
import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

// Local imports
import Image from '../../Components/Atoms/Image';
import { blankAvatar } from '../../images';

describe('<Image />', () => {
  it('renders and match the snapshot', () => {
    const { asFragment } = render(
      <Image imageAlt="testAlt" className="testClass" image={blankAvatar} />
    );

    expect(asFragment).toMatchSnapshot();
  });

  it('has properly className', () => {
    render(<Image className="testClass" image={blankAvatar} />);
    const imageContainer = screen.getByTestId('atoms-image');

    expect(imageContainer).toHaveClass('testClass');
  });

  it('has properly alt attribute', () => {
    render(<Image imageAlt="test-alt" image={blankAvatar} />);
    const imageContainer = screen.getByTestId('atoms-image');

    expect(imageContainer).toHaveAttribute('alt', 'test-alt');
  });

  it('has properly background image', () => {
    const image = blankAvatar;
    render(<Image image={image} />);
    const imageContainer = screen.getByTestId('atoms-image');

    expect(imageContainer).toHaveStyle(`background-image: url(${image})`);
  });
});
