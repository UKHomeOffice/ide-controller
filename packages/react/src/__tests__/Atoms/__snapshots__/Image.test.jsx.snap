// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Image /> renders and match the snapshot 1`] = `[Function]`;
